### PR TITLE
Apply naming convention to workflows

### PR DESCRIPTION
**Pull request by Link-**
### Context

After introducing the new autoscaling runner scale set mode as well as the new helm charts and because we didn't want to break the existing release process we opted to create a new set of workflows to publish the new mode.

This is a housekeeping PR to cleanup the workflow file names and workflow names to be clearer and reduce operational confusion.

Fixes: https://github.com/github/c2c-actions-runtime/issues/2396

| original filename             | original workflow name                                                         | new filename                | new workflow name              |
|-------------------------------|--------------------------------------------------------------------------------|-----------------------------------|--------------------------------------|
| e2e-test-linux-vm.yaml        | CI ARC E2E Linux VM Test                                                       | gha-e2e-tests.yaml                | (gha) E2E Tests                      |
| go.yaml                       | Go                                                                             | go.yaml                           | Format, Lint, Unit Tests             |
| publish-arc.yaml              | Publish ARC                                                                    | arc-publish.yaml                  | Publish ARC Image                         |
| publish-canary.yaml           | Publish Canary Image                                                           | global-publish-canary.yaml           | Publish Canary Images             |
| publish-chart.yaml            | Publish Helm Chart                                                             | arc-publish-chart.yaml            | Publish ARC Helm Charts              |
| publish-runner-scale-set.yaml | Publish Runner Scale Set Controller Charts                                     | gha-publish-chart.yaml            | (gha) Publish Helm Charts            |
| release-runners.yaml          | Runners                                                                        | arc-release-runners.yaml          | Release ARC Runner Images            |
| run-codeql.yaml               | Run CodeQL                                                                     | global-run-codeql.yaml            | Run CodeQL                           |
| run-first-interaction.yaml    | first-interaction                                                              | global-run-first-interaction.yaml | First Interaction                    |
| run-stale.yaml                | Run Stale Bot                                                                  | global-run-stale.yaml             | Run Stale Bot                        |
| update-runners.yaml           | Update runners                                                                 | arc-update-runners-scheduled.yaml | Runner Updates Check (Scheduled Job) |
| validate-chart.yaml           | Validate Helm Chart                                                            | arc-validate-chart.yaml           | Validate Helm Chart                  |
| validate-gha-chart.yaml       | Validate Helm Chart (gha-runner-scale-set-controller and gha-runner-scale-set) | gha-validate-chart.yaml           | (gha) Validate Helm Charts           |
| validate-runners.yaml         | Validate Runners                                                               | arc-validate-runners.yaml         | Validate ARC Runners                 |

### CODE REVIEW

**Comment by Link-** [comment]
This release needs to go out before this PR is merged: https://github.com/actions/actions-runner-controller/pull/2577

**Review by benwells — DISMISSED** [review]

**Review by nikola-jokic — DISMISSED** [review]

**Comment by Link-** [comment]
I'll merge this on Monday because I suspect it'll break a bunch of stuff and I want the time buffer to handle them

File: .github/workflows/arc-publish-canary.yaml [hunk 1]
```diff
@@ -11,19 +11,19 @@ on:
...
-      - '.github/workflows/publish-arc.yaml'
-      - '.github/workflows/publish-chart.yaml'
-      - '.github/workflows/publish-runner-scale-set.yaml'
-      - '.github/workflows/release-runners.yaml'
-      - '.github/workflows/run-codeql.yaml'
-      - '.github/workflows/run-first-interaction.yaml'
-      - '.github/workflows/run-stale.yaml'
-      - '.github/workflows/update-runners.yaml'
+      - '.github/workflows/gha-e2e-tests.yaml'
+      - '.github/workflows/arc-publish.yaml	'
```
Comment (johnsudol) [inline .github/workflows/arc-publish-canary.yaml]: nit
```suggestion
      - '.github/workflows/arc-publish.yaml'
```

Comment (Link-) [reply .github/workflows/arc-publish-canary.yaml]: Could you redo it please? I'm not able to merge this one cause I made a change before seeing this

File: .github/workflows/global-publish-canary.yaml [hunk 2]
```diff
@@ -11,19 +11,19 @@ on:
...
-      - '.github/workflows/publish-arc.yaml'
-      - '.github/workflows/publish-chart.yaml'
-      - '.github/workflows/publish-runner-scale-set.yaml'
-      - '.github/workflows/release-runners.yaml'
-      - '.github/workflows/run-codeql.yaml'
-      - '.github/workflows/run-first-interaction.yaml'
-      - '.github/workflows/run-stale.yaml'
-      - '.github/workflows/update-runners.yaml'
+      - '.github/workflows/gha-e2e-tests.yaml'
+      - '.github/workflows/arc-publish.yaml	'
```
Comment (johnsudol) [inline .github/workflows/global-publish-canary.yaml]:
```suggestion
      - '.github/workflows/arc-publish.yaml'
```

**Comment by Link-** [comment]
The fix for: `(gha) Validate Helm Charts / Lint Chart (pull_request)` is in: https://github.com/actions/actions-runner-controller/pull/2569

**Review by nikola-jokic — APPROVED** [review]

**Review by rentziass — APPROVED** [review]

**Pull request merged**